test/tests/functional/pbs_ralter.py

```diff
@@ -1516,3 +1548,265 @@ def test_failed_ralter(self):
+                                 check_log=False)
+
+        t_duration, t_start, t_end = self.get_resv_time_info(rid)
+
+        self.assertEqual(t_end, t_start+t_duration)
+        self.assertEqual(t_start, start+shift)
+        self.assertEqual(t_duration, new_duration)
+
+        # Submit a job to the reservation and change its start time.
+        self.submit_job_to_resv(rid)
+        temp_end = t_end
+        temp_start = t_start
+
+        new_duration2 = new_duration + 10
+        self.alter_a_reservation(rid, temp_start, temp_end, shift=shift,
+                                 alter_s=True, sequence=2,
+                                 a_duration=new_duration2, check_log=False)
+        t_duration, t_start, t_end = self.get_resv_time_info(rid)
+        self.assertEqual(t_end, t_start+t_duration)
+        self.assertEqual(t_start, temp_start+shift)
+        self.assertEqual(t_duration, new_duration2)
+
+    def test_adv_res_dur_after_start(self):
+        """
+        Test that duration can be changed after the reservation starts.
+        Test that if the duration changes endtime of the reservation to an
+        already passed time, the reservation is deleted
+        """
+        offset = 10
+        duration = 20
+        new_duration = 30
+        shift = 10
+        rid, start, end = self.submit_and_confirm_reservation(offset, duration)
+
+        self.check_resv_running(rid, offset)
+
+        self.alter_a_reservation(rid, start, end, a_duration=new_duration,
+                                 confirm=False, check_log=False)
+
+        t_duration, t_start, t_end = self.get_resv_time_info(rid)
+        self.assertEqual(t_duration, new_duration)
+
+        time.sleep(5)
+        attr = {'reserve_duration': 5}
+        self.server.alterresv(rid, attr)
+        self.server.log_match(rid + ";Reservation alter denied",
+                              id=rid, interval=2)
+        rid = rid.split('.')[0]
+        self.server.log_match(rid + ";deleted at request of pbs_server",
+                              id=rid, interval=2)
+
+    def test_adv_resv_endtime_starttime_dur_together(self):
+        """
+        Test that all three end, start and duration cannot be changed together
+        """
+        offset = 20
+        duration = 20
+        new_duration = 30
+        shift = 10
+        rid, start, end = self.submit_and_confirm_reservation(offset, duration)
+        new_start = self.bu.convert_seconds_to_datetime(start + shift)
+        new_end = self.bu.convert_seconds_to_datetime(end + shift)
+        new_duration = self.bu.convert_seconds_to_datetime(new_duration)
+
+        with self.assertRaises(PbsResvAlterError) as e:
+            attr = {'reserve_start': new_start, 'reserve_end': new_end,
+                    'reserve_duration': new_duration}
+            self.server.alterresv(rid, attr)
+            self.assertIn('pbs_ralter: Bad time specification(s)', e.msg[0])
```

Comment:
This block of code will end when `self.server.alterresv` throws the reservation, so this line needs to be outside the with block.

Also, make sure that the reservation still has the original start/end/dur.